### ISSUE TITLE
Targeting .net-4.5 results in being stuck with ancient compiler only supporting csharp-6

### ISSUE THREAD

**Issue by binki** [aspnet/RoslynCodeDomProvider#61]
If I install `Microsoft.CodeDom.Providers.DotNetCompilerPlatform` into a project targeting .net-4.5.2 with `LangVersion` set to `latest`, I get this error when my project compiles:

> CS1617 Invalid option 'latest' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.

I can easily target .net-4.5.2 with a modern compiler when not using `Microsoft.CodeDom.Providers.DotNetCompilerPlatform`—i.e., when using visualstudio’s built-in compiler. However, I need to use `Microsoft.CodeDom.Providers.DotNetCompilerPlatform` in this project to support ASP.NET.

Please allow using `tools/RoslynLatest` with .net-4.5.x in a non-hacky way or just make it the default. I don’t see what benefit there is to having `tools/Roslyn45` at all because most people would need very specific versions of roslyn if they’re dodging [compiler](https://developercommunity.visualstudio.com/content/problem/315090/.html) [crashes](https://developercommunity.visualstudio.com/content/problem/30649/.html) or [regressions](https://github.com/dotnet/roslyn/issues/21028) rather than the choice between the latest csharp-7 and csharp-6 variants.

I’m using visualstudio-15.8.3. When I read https://github.com/aspnet/RoslynCodeDomProvider/blob/12975a5a1616dbefac02ba8113204b59cba7eeec/README.md , I am lead to believe that the package should be using the roslyn provided by visualstudio during compilation. But it appears it is actually using its own provided roslyn instead.

Even if it were using visualstudio’s roslyn during project compilation, it still needs to copy the correct tool to `bin` so that ASP.NET can get `tools/RolsynLatest` instead of `tools/Rolsyn45`. If it can copy from visualstudio, then that would be fine, though.

I am considering working around this by doing:

```xml
<PropertyGroup>
  <HOME Condition="'$(HOME)' == ''">$(USERPROFILE)</HOME>
  <RoslynToolPath>$(HOME)/.nuget/packages/microsoft.codedom.providers.dotnetcompilerplatform/2.0.0/tools/RoslynLatest</RoslynToolPath>
</PropertyGroup>
```

but that doesn’t seem very portable and relies on internal implementation details of nuget. It does at least fix the issue on my machine with its default nuget setup.

**Comment by Jinhuafei** [comment]
@binki can you check if your project still has Microsoft.Net.Compilers nupkg? Microsoft.Net.Compilers nupkg will make msbuild to use the Roslyn compiler from the Microsoft.Net.Compilers instead of the one shipped with VS.

**Comment by binki** [comment]
@Jinhuafei When I was experiencing the problem, I tried adding `Microsoft.Net.Compilers-2.9.0` because I hoped that it would override the Roslyn compiler with one that could handle `/langversion:latest`. But that didn’t change anything.

However, now when I try to reproduce this issue, I don’t see that happening at all. In fact, I can’t get the issue to happen again, even after trying a few things. Right now, if I use versions 2.0.0 or 2.0.1 of this nupkg while targeting .net-4.5.2, the build will use VS’s `csc.exe` which lets me use modern C# and `/langversion:latest` in the precompiled portions of my project while copying the .net-4.5 version of roslyn to `bin` restricting my `.aspx` files to C#-6.

This is in a somewhat complex VS solution with a lot of projects and some dependencies. It also was not fully building at the time and I was transitioning from `packages.config` to `<PackageReference/>`. I don’t think I’ll be able to track down the weird behavior I was experiencing originally.

Because of that, I don’t think my issue makes any sense anymore. I learned (while filing this issue) that the nuget will purposefully install the last version of roslyn which supports .net-4.5 when targeting .net-4.5. It makes no sense to change that behavior: if I am really deploying my webapp to a system which only supports .net-4.5, then it won’t be capable of running the bundled roslyn to compile `.aspx` files.

In my case, with the 1.x series, I was able to have my project target .net-4.5 but also depend on something like Microsoft.Net.Compilers-2.3.0 and it let me do that. When I actually deployed, the server must’ve already been running .net-4.6+ already, otherwise we would’ve hit issues.

Sorry for the false bug report and thanks for getting back with me!
